src/compressor/number.ts

```diff
@@ -16,7 +16,7 @@ export function compressNumber(
 ) {
   let foundRef: string | undefined;
 
-  if(obj % 1 === 0) {
+  if(Number.isInteger(obj) && obj.toString() !== obj.toExponential()) {
```

Comment:
I don't think you have to change the existing condition,
`obj % 1 === 0 && obj.toString() !== obj.toExponential()`
Should also work I think?

Reply:
Thought inner implementation should be more performant... But i see that according to docs it may return wrong result with high precision. Maybe you're right. Will revert this

Reply:
> Thought inner implementation should be more performant... But i see that according to docs it may return wrong result with high precision. Maybe you're right. Will revert this

Yeah could be, hard to tell unless benchmarked. I’m mostly thinking just for compatibility